From 5d9c9aa9ec6031e130f210b04ae32de7aa9e85b8 Mon Sep 17 00:00:00 2001
From: Christian Hohnstaedt <christian@hohnstaedt.de>
Date: Fri, 17 Feb 2023 23:36:24 +0100
Subject: [PATCH] execline: multicall: make sort independent of locale

reset LC_COLLATE to avoid locale depndent sorting.

man page of "sort" states:

***  WARNING  *** The locale specified by the environment affects sort order.
Set LC_ALL=C to get the traditional sort order that uses native byte values.
---
 tools/gen-multicall.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/gen-multicall.sh b/tools/gen-multicall.sh
index 1d3d9fe..1f7b6a1 100755
--- a/tools/gen-multicall.sh
+++ b/tools/gen-multicall.sh
@@ -59,7 +59,7 @@ static execline_app const execline_apps[] =
 {
 EOF
 
-for i in `{ echo cd ; echo execline ; echo umask ; ls -1 src/execline/deps-exe ; } | sort` ; do
+for i in `{ echo cd ; echo execline ; echo umask ; ls -1 src/execline/deps-exe ; } | LC_ALL=C sort` ; do
   j=`echo $i | tr - _`
   if test $i = cd ; then
     echo '  { .name = "cd", .mainf = (main_func_ref)&CD_FUNC },'
-- 
2.30.2

